scipy/stats/_multivariate.py

```diff
@@ -786,6 +786,36 @@ def entropy(self, mean=None, cov=1):
         dim, mean, cov_object = self._process_parameters(mean, cov)
         return 0.5 * (cov_object.rank * (_LOG_2PI + 1) + cov_object.log_pdet)
 
+    def fit(self, x):
+        """Fit the multivariate normal to data `x`.
+
+        Parameters
+        ----------
+        x : data the distribution is fitted to. The last axis determines the
+            dimension of the fitted distribution.
+
+        Returns
+        -------
+        mean : Estimated mean vector
+        cov : Estimated covariance matrix
+
+        """
+        # input validation
+        x = np.asarray(x)
+        if x.ndim < 2:
+            raise ValueError("`x` must be at least two-dimensional.")
+
+        # flatten array so that vectors are in last axis
+        dim = x.shape[-1]
+        n_vectors = math.prod(x.shape[:-1])
+        x = x.reshape((n_vectors, dim))
```

Comment:
```suggestion
        x = x.reshape((-1, dim))
```
But actually, I would prefer to raise an error if `ndim!=2`. 

You may have been following the behavior of `stats.norm.fit`, which allows multivariate input and implicitly ravels it. But I think that is a mistake in the override. `stats.rv_continuous.fit` raises if the input is not 1d. 

https://github.com/scipy/scipy/blob/c1ed5ece8ffbf05356a22a8106affcd11bd3aee0/scipy/stats/_fit.py#L548-L551

(Of course, we should really only be overriding a `_fit` method so that this sort of input validation can be handled by `fit`, but that is a different story.)

Anyway, raising an error now lets us be more flexible in the future if, for example, we want to begin to allow fitting independently over multiple data sets in one vectorized call. (If we were to reshape like this here, the default `axis` would have to be `None`.)

Reply:
In the recent PR for the von Mises Fisher distribution (#17624), I did not limit the dimensions for data to be fitted to 2D.  For simplicity of the API, that makes sense though and be more consistent with the univariate distribution. As that one is already merged, shall we first change it there before 1.11 is released?